### DISCUSSION TITLE
How do I set DenyAddAndCustomizePagesStatus disabled with C# ?

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/pnpcore#1154] [Q&A] [answered]
I have code to add a property bag value with C# using PnP.Core (1.9)
```
using (var newSiteContext = await pnpContext.GetSiteCollectionManager().CreateSiteCollectionAsync(noTeamSiteToCreate))
{
	// Load the web property bag
	var web = await newSiteContext.Web.GetAsync(p => p.AllProperties);
	// Add a new property SiteType
	web.AllProperties["SiteType"] = "Client Site No Team";
	await web.AllProperties.UpdateAsync();
	return newSiteContext.Uri.AbsoluteUri;
}
```
it returns an error "CSOM service exception" "Message: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))"
I assume it is because of the "No custom code allowed" of modern site.
In PowerShell I would do:
```
Set-PnPSite -NoScriptSite $false -Connection $projectConn  #-DenyAddAndCustomizePages 0
Set-PnPPropertyBagValue -Key "SiteType" -Value "Client Site with Team" -Indexed -Connection $projectConn
Set-PnPSite -NoScriptSite $true -Connection $projectConn  #-DenyAddAndCustomizePages 1			
```
But I can't figure out the PnP.Core equivalent

Can anyone help here?

Thank you

**Comment by jansenbe** [comment, marked as answer]
@Ofer-Gal : In PnP Core SDK this can be done using the admin library by setting the [DenyAddAndCustomizePages](https://pnp.github.io/pnpcore/api/PnP.Core.Admin.Model.SharePoint.ISiteCollectionProperties.html#collapsible-PnP_Core_Admin_Model_SharePoint_ISiteCollectionProperties_DenyAddAndCustomizePages) property of an `ISiteCollection`. To learn how to set a site collection property see https://pnp.github.io/pnpcore/using-the-sdk/admin-sharepoint-sites.html#getting-and-setting-site-collection-properties

**Reply by Ofer-Gal** [reply]
That worked fine.
Thank you very much.
